### PR TITLE
Add shields for Puerto Rico

### LINKED ISSUE #240: Puerto Rico shields
**(EN)**

Puerto Rico has only one network of signposted routes, though signage can shift between any of 4 shields depending on a segment's importance in the road hierarchy. Route numbers can have up to 4 digits.

Current coverage of `route=road` relations in Puerto Rico is incomplete, and existing relations are tagged with `network=US:PR`, conflating all shields into one network. There has been no discussion yet towards new `network` values that correspond to shield appearance.

**(ES)**

Puerto Rico tiene una sola red de rutas rotuladas, aunque la rotulación puede cambiar entre cualquiera de los 4 diseños según la importancia de un segmento en la jerarquía vial. Los números de las rutas pueden tener hasta 4 dígitos.

La cobertura actual de las relaciones `route=road` es incompleta, y las relaciones existentes tienen la etiqueta `network=US:PR`, combinando todos los diseños en una red. Todavía no ha habido discusión hacia nuevos valores de la clave `network` que correspondan a diseños de identificadores de rutas. 

----

Primary shield / identificador primario:

Puerto Rico primary shield

Urban primary shield / identificador primario urbano:

Puerto Rico urban primary shield

Secondary shield / identificador secundario:

Puerto Rico secondary shield

Tertiary shield / identificador terciario:

Puerto Rico tertiary shield

### PR DESCRIPTION
**Pull request by claysmalley**
Fixes #240.

Waiting on a tile update before screenshots can be produced.

### CODE REVIEW

**Review by 1ec5 — COMMENTED** [review]
This is also waiting on confirmation of [my tagging proposal](https://lists.openstreetmap.org/pipermail/talk-pr/2022-July/000393.html) on the talk-pr list. The relations I added in https://github.com/osm-americana/openstreetmap-americana/issues/240#issuecomment-1184802189 are for demonstration purposes only so far.

**Comment by ZeLonewolf** [comment]
Setting draft pending satisfaction of local community engagement.

**Comment by 1ec5** [comment]
From the experiment in https://github.com/ZeLonewolf/openstreetmap-americana/issues/240#issuecomment-1184802189:

[Guayama](https://zelonewolf.github.io/openstreetmap-americana/#14/17.97534/-66.11473)

**Comment by 1ec5** [comment]
> This is also waiting on confirmation of [my tagging proposal](https://lists.openstreetmap.org/pipermail/talk-pr/2022-July/000393.html) on the talk-pr list.

The local community is [on board](https://lists.openstreetmap.org/pipermail/talk-pr/2022-July/000400.html), with one modification to replace `network=US:PR:urban_primary` with `network=US:PR:primary_urban` for resemblance to Spanish. [Changeset 124,081,899](https://www.openstreetmap.org/changeset/124081899) takes care of this change, but it’ll require another refresh of the vector tiles. This gives us some time to tag or retag more routes in the meantime.

File: README.md [hunk 1]
```diff
@@ -48,12 +48,8 @@ The OpenStreetMap Americana style is built upon the [OpenMapTiles schema](https:
...
-
-<img src="doc-img/shield_map_us.svg" width="500" alt="U.S. states">
-
-and for routes in the following countries:
+Americana displays custom route shields for routes in all U.S. states and territories, and for routes in the following countries:
 
 <img src="doc-img/shield_map_world.svg" width="500" alt="Countries">
 
-We are hoping that it will support more areas, you can [help us](CONTRIBUTING.md)!
+We are hoping that it will support more countries; you can [help us](https://github.com/ZeLonewolf/openstreetmap-americana/projects/1)!
```
[the remark this answers is not in the document]
Comment (claysmalley) [reply README.md:55]: There is already a link to CONTRIBUTING.md above, and this sentence is specifically about shield internationalization.

File: scripts/status_map.js [hunk 2]
```diff
@@ -33,22 +28,6 @@ let shieldImageURLs = Object.fromEntries(
...
-    .map((network) => network.match(/^US:(\w\w)(?::|$)/))
-    .filter((m) => m)
-    .map((m) => m[1])
-);
-usSVG = usSVG.replace(
-  /<title>.+?<\/title>/,
-  "<title>U.S. states with shields supported by OpenStreetMap Americana</title>"
-);
-fs.writeFileSync(`${process.cwd()}/doc-img/shield_map_us.svg`, usSVG);
-
```
Comment (ZeLonewolf) [inline scripts/status_map.js:51]: Is this code removed because we no longer need a US map?

Comment (claysmalley) [reply scripts/status_map.js:51]: Yep. With Puerto Rico added, the US map is complete so it's no longer necessary.

**Comment by ZeLonewolf** [comment]
Code looks essentially correct.  I visually confirmed that HK, and US:VA shields are displaying as expected.

I was not able to locate a road in PR with the intended shields.  I recall from memory that the main road from San Juan to Farjardo was shielded but didn't find shields displaying either there or other places I looked on the island.

**Comment by ZeLonewolf** [comment]
Definitely seeing some tagging issues, e.g. in the area around Arecibo:

image

**Comment by ZeLonewolf** [comment]
In the area around Rincon, on the west coast of the island:

image

**Comment by claysmalley** [comment]
Here's Guayama: [(localhost)](http://localhost:1776/#16.03/17.974419/-66.109672)

Puerto Rico's situation is similar to that of Tennessee. The majority of routes have not yet been mapped to distinguish segments with different shield appearance.

**Comment by ZeLonewolf** [comment]
Linked location looks good to me.  An overpass query of `network=US:PR` comes up with a fair number of hits on relations that don't seem to correspond with shields and could use a look:

http://overpass-turbo.eu/?Q=%2F*%0AThis%20is%20an%20example%20Overpass%20query.%0ATry%20it%20out%20by%20pressing%20the%20Run%20button%20above!%0AYou%20can%20find%20more%20examples%20with%20the%20Load%20tool.%0A*%2F%0Arel%5Bnetwork%3D%22US%3APR%22%5D%3B%0A(._%3B%3E%3B)%3B%0Aout%3B&C=18.27045;-66.45719;10

For example:
https://www.openstreetmap.org/relation/3754823

Looks like the refs include the whole network in them which overflows our ref length limit.

**Pull request merged**